app/src/main/java/org/mifos/mobile/ui/transaction/TransactionActivity.kt

```diff
@@ -0,0 +1,87 @@
+package org.mifos.mobile.ui.transaction
+
+import android.os.Bundle
+import android.text.TextUtils
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import kotlinx.android.synthetic.main.activity_transaction.*
+import org.json.JSONObject
+import org.mifos.mobile.R
+import org.mifos.mobile.models.entity.*
+import org.mifos.mobile.ui.activities.base.BaseActivity
+import org.mifos.mobile.utils.Toaster
+import javax.inject.Inject
+
+class TransactionActivity : BaseActivity(), TransactionContract.View, View.OnClickListener {
+
+    @Inject
+    internal lateinit var transactionPresenter: TransactionPresenter
+    var jsonob: JSONObject? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_transaction)
+        showBackButton()
+        setToolbarTitle("Send Money")
+        activityComponent?.inject(this)
+        transactionPresenter.attachView(this)
+        send.setOnClickListener(this)
+        if(intent.getStringExtra("JSON")!=null) {
+            jsonob = JSONObject(intent.getStringExtra("JSON"))
+            MSISDNtext.setText(jsonob?.getString("mobile"))
+        }
+    }
+
+    override fun onClick(view: View) {
+
+        when (view.id) {
+            R.id.send -> transaction()
+        }
+
+    }
+
+    fun transaction() {
+
+        val amount = amounttext.text.toString().trim()
+        val msisdn = MSISDNtext.text.toString().trim()
+        if (TextUtils.isEmpty(amount)) {
+            amounttext.error = "Amount required"
+            return
+        }
+        if (TextUtils.isEmpty(msisdn)) {
+            MSISDNtext.error = "MSISDN required"
+            return
+        }
+
+        val transaction = Transaction(TransactingEntity
+        (PartyIdInfo(IdentifierType.MSISDN, "76894434000")),
```

Comment:
why are we hardcoding?`76894434000` and `TZS`

Reply:
Currently, the registration use case is not yet implemented. Without that, we cannot use the PaymentHub APIs for the payer, because the user is not registered in the Payment Hub. Once registered, we then have to fetch the secondary identifiers from the APIs for the `payer` and then use his/her `MSISDN` id in order to make the transaction. I am planning to implement the registration use case in the next PR. Till then, we are just using the dummy credentials (`MSISDN` id ) of an already registered user to test the integration.